### PR TITLE
Three bug fixed concerning target termination on linux

### PR DESCRIPTION
**Pull request by aarzilli**
See commit messages for a description. I'm not sure about the last one, it's a workaround, I don't know why that happens.

### CODE REVIEW

**Comment by derekparker** [comment]
Can you rebase this on master? The changes look good, however.

**Comment by aarzilli** [comment]
done

File: proc/proc.go [hunk 1]
```diff
@@ -101,11 +101,13 @@ func (dbp *Process) Detach(kill bool) (err error) {
 		}
 	}
 	dbp.execPtraceFunc(func() {
-		var sig int
+		err = PtraceDetach(dbp.Pid, 0)
```
Comment (derekparker) [inline proc/proc.go]: We should return here upon error, so it doesn't get masked below.

Comment (aarzilli) [inline proc/proc.go]: It doesn't, if kill isn't set the func ends, if it is and err is not nil it returns. But it's convoluted, I've changed it.

**Pull request merged**